### PR TITLE
Fix SSL on Mac OS X with GraalVM 1.0.0-rc14

### LINKED ISSUE #1665: SSL broken on MacOS X with GraalVM 1.0.0-rc14
The `libsunec.dylib` is not in the amd64 directory anymore but in the parent. This is a changed introduced in the last GraalVM release. Note that the location didn't change for linux.

### PR DESCRIPTION
**Pull request by cescoffier**
Improved a bit the java.library.path detection. It's still pretty naive, but works.
The real issue is that substrateVM does not allow multiple elements in the java.library.path, so we must pick one.
The detection is based on the number of files in the amd64 directory. However this is definitely brittle, and if the next versions of GraalVM add files into this directory, we would need to increment the number.

This fixes #1665

### CODE REVIEW

**Comment by cescoffier** [comment]
@Sanne can you check on Fedora?

I've all the quickstarts passing.

File: core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java [hunk 1]
```diff
@@ -88,16 +91,16 @@ void build(List<SubstrateConfigBuildItem> substrateConfigBuildItems,
...
             if (graalVmHome != null) {
                 Path graalVmLibDirectory = Paths.get(graalVmHome, "jre", "lib");
-                Path linuxLibDirectory = graalVmLibDirectory.resolve("amd64");
+                Path amd64LibDirectory = graalVmLibDirectory.resolve("amd64");
 
                 // We add . as it might be useful in a containerized world
                 // FIXME: it seems GraalVM does not support having multiple paths in java.library.path
                 //javaLibraryPathAdditionalPath.produce(new JavaLibraryPathAdditionalPathBuildItem("."));
-                if (Files.exists(linuxLibDirectory)) {
+                if (existsAndDoesNotContainOnlyTheAgent(amd64LibDirectory)) {
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java]: Why don't we test the existence of the files? We could do something like:
- have a list of **file** candidates
- if exists, add the directory of the file to the library path

**Review by gsmet — CHANGES_REQUESTED** [review]
I definitely don't like the approach as it's even more brittle than it used to be.

I proposed another approach inline. Could you give it a shot?

**Review by kenfinnigan — APPROVED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java (hunk 1 above)
Comment (cescoffier) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java]: The issue is that creates a tight dependency on a specific SSL related file, which is not great either.

Comment (cescoffier) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java]: What's the exact file name on Linux?

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java]: `/opt/graalvm/jre/lib/amd64/libsunec.so`

Maybe we could visit the `jre/lib/` hierarchy and try to find a file containing `sunec` (or starting with `libsunec`, I don't know the macOS name) in its name and then select the directory?

Comment (cescoffier) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java]: Wondering about windows, so right now I'm working on finding the Linux one.

Comment (cescoffier) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java]: @gsmet Can you check https://github.com/quarkusio/quarkus/pull/1681.

**Comment by gsmet** [comment]
I merged the other one.

**Pull request closed without merging**